arelle/plugin/validate/ROS/rules/__init__.py

```diff
@@ -17,16 +18,41 @@ def errorOnRequiredFact(
         message: str,
 ) -> Iterable[Validation]:
     """
-    Yields an error if a fact with the given QName is not tagged with a non-nil value.
+    Yields an error if a fact with the given localName is not tagged with a non-nil value.
     :return: Yields validation errors.
     """
     facts: set[ModelFact] = modelXbrl.factsByLocalName.get(conceptLn, set())
     for fact in facts:
         if not fact.isNil:
             return
     yield Validation.error(
+        conceptLn=conceptLn,
         codes=code,
         msg=message,
-        conceptLn=conceptLn,
+
 
     )
+
+
+def errorOnPositiveOnlyFact(
```

Comment:
I think the name here is the opposite of the implementation. It errors on a negative fact. Suggest `errorOnNegativeFact`.